src/carousel/index.ts

```diff
@@ -0,0 +1,283 @@
+    }
+
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+
+    let itemCount = this.props.items.length;
+    let preCloneSlides = [];
+    let postCloneSlides = [];
+
+    this.props.items.forEach((data, index) => {
+      data['data-index'] = index;
+
+      if (index >= (itemCount - slidesToShow)) {
+        let key = -(itemCount - index);
+        preCloneSlides.push({
+          ...data,
+          'data-index': key,
+        });
+      } else if (index < slidesToShow) {
+
+        let key = itemCount + index;
+        postCloneSlides.push({
+          ...data,
+          'data-index': key,
+        });
+      }
+    });
+
+    return preCloneSlides.concat(this.props.items, postCloneSlides);
+  }
+
+  goToSlide = (slide: number) => {
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const from = this.currentSlide;
+    const to = slide;
+
+    // make the transition
+    this.currentSlide = slide;
+
+    const threshold = this.props.items.length;
+    const rightBound = this.currentSlide + slidesToShow - 1;
+
+    const listener = () => {
+      if (from < to) {
+        this.currentSlide = this.currentSlide - threshold;
+      } else {
+        this.currentSlide = this.currentSlide + threshold;
+      }
+      this.refs.track.removeEventListener('transitionend', listener);
+      this.temporaryNoTransition = true;
+      this.update();
+    };
+
+    if (rightBound >= threshold || this.currentSlide < 0) {
+      this.refs.track.addEventListener('transitionend', listener);
+    }
+  }
+
+  goToDot = (e: MouseEvent | TouchEvent) => {
+    e.preventDefault();
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+
+    const slide = parseInt((e.target as HTMLElement).getAttribute('data-index-to-go')) * slidesToShow;
+    this.goToSlide(slide);
+  }
+
+  getSlideStyle = () => {
+    const slideWidth = this.getSlideWidth();
+    return {
+      width: `${slideWidth}px`,
+      outline: 'none',
+    };
+  }
+
+  getTrackStyle = () => {
+    if (!(this.cloneItems() && this.props.items)) {
+      return;
+    }
+    const slideWidth = this.getSlideWidth();
+    const slideCount = this.cloneItems().length;
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const transition = typeof this.props.settings.transition === 'boolean' ? 
+      this.props.settings.transition : DEFAULT_SETTINGS.transition;
+    const speed = this.props.settings.speed || DEFAULT_SETTINGS.speed;
+
+    const trackWidth = (slideCount + 2 * slidesToShow) * slideWidth;
+    const pos = this.calcPos(this.currentSlide, slideWidth);
+    const tfm = `translate3d(-${pos}px, 0px, 0px)`;
+
+    const transformStyles = {
+      transform: tfm,
+      '-webkit-transform': tfm,
+      '-ms-transform': tfm,
+    };
+
+    const tsVal = speed + 'ms ' + 'ease';
+    const transitionStyles = transition === true ? {
```

Comment:
Don't have to check true === true or false === true can just do `transition ? ...`